shared/js/background/amp-protection.js

```diff
@@ -48,8 +48,23 @@ function extractAMPURL (site, url) {
 
     for (const regexPattern of ampSettings.linkFormats) {
         const match = url.match(regexPattern)
+
+        let newUrl
         if (match && match.length > 1) {
-            const newSite = new Site(match[1].startsWith('http') ? match[1] : `https://${match[1]}`)
+            try {
+                newUrl = new URL(match[1])
+                if (!newUrl.protocol.startsWith('http')) {
+                    return null
+                }
+            } catch (e) {
+                try {
+                    newUrl = new URL(`https://${match[1]}`)
+                } catch (e) {
```

Comment:
You don't need e here at all for either
```suggestion
                } catch {
```